crates/bevy_ecs/src/schedule/executor/mod.rs

```diff
@@ -303,6 +305,16 @@ mod __rust_begin_short_backtrace {
         result
     }
 
+    #[inline(never)]
+    pub(super) fn run_without_applying_deferred(
+        system: &mut ScheduleSystem,
+        world: &mut World,
+    ) -> Result {
+        let result = system.run_without_applying_deferred((), world);
+        black_box(());
```

Comment:
This is the first time I've seen `black_box` used like this, and after some searching I still don't know why this would be beneficial. Can you explain or link me to somewhere this is explained?